### PR TITLE
feat(storage): GA gRPC client 

### PR DESCRIPTION
**Pull request by frankyn**
- [ ] Pending zero-copy improvements from @tritone

### CODE REVIEW

File: storage/doc.go [hunk 1]
```diff
@@ -331,14 +331,12 @@ to add a [custom audit logging] header:
...
+# gRPC API
 
-This package includes support for the Cloud Storage gRPC API, which is currently
-in preview. This implementation uses gRPC rather than the current JSON & XML
-APIs to make requests to Cloud Storage. Kindly contact the Google Cloud Storage gRPC
-team at gcs-grpc-contact@google.com with a list of GCS buckets you would like to
-allowlist to access this API. The Go Storage gRPC library is not yet generally
-available, so it may be subject to breaking changes.
+This package includes support for the Cloud Storage gRPC API. The
+implementation uses gRPC rather than the current
```
Comment (tritone) [inline storage/doc.go:337]: s/current/default

File: storage/doc.go [hunk 2]
```diff
@@ -331,14 +331,12 @@ to add a [custom audit logging] header:
...
+JSON & XML APIs to make requests to Cloud Storage.
+The Go Storage gRPC library is generally available.
```
Comment (tritone) [inline storage/doc.go:339]: s/library/client.

File: storage/doc.go [hunk 3]
```diff
@@ -349,11 +347,12 @@ To create a client which will use gRPC, use the alternate constructor:
 	}
 	// Use client as usual.
 
-Using the gRPC API inside GCP with a bucket in the same region can allow for
-Direct Google Access (enabling requests to skip some proxy steps and reducing
-response latency). A warning is emmitted if gRPC is not used within GCP to
-warn that Direct Google Access could not be initialized. Direct Google Access
-is not required to access the gRPC API.
+Using the gRPC API inside GCP with a bucket in the same region can allow direct
+connectivity (enabling requests to skip Google Front Ends (GFEs) and reducing
```
Comment (tritone) [inline storage/doc.go:351]: Google Front Ends is another term without public docs, I would revert to the original wording here and just add the link for the (new?) DP docs. No need to duplicate more information here.

**Review by tritone — COMMENTED** [review]
Few docs nits, otherwise looks good. Let's wait to submit this until after zerocopy changes have been merged.

**Review by codyoss — COMMENTED** [review]
Should we make this a feat so it shows up in release notes as a new feature?

**Comment by frankyn** [comment]
Thanks @codyoss updated PR name to `feat`; I think it makes sense so it's in release notes.

**Review by tritone — APPROVED** [review]

**Pull request merged**